hack/jenkins/e2e.sh

```diff
@@ -190,6 +204,194 @@ case ${JOB_NAME} in
     MINION_SIZE="n1-standard-1"
     NUM_MINIONS="2"
     ;;
+
+  # GCE master+node upgrade step 1: deploy a cluster at latest release.
+  kubernetes-upgrade-gce-step1-deploy)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_DOWN:="false"}
+    : ${E2E_TEST:="false"}
+    : ${JENKINS_USE_RELEASE_TARS:="y"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GCE master+node upgrade step 2: upgrade the cluster from step 1 to the
+  # latest CI version using the latest CI verion's upgrade code.
+  kubernetes-upgrade-gce-step2-upgrade)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_DOWN:="false"}
+    : ${E2E_OPT:="--check_version_skew=false"}
+    : ${E2E_UP:="false"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.focus=Skipped.*Cluster\supgrade.*upgrade-cluster"}
+    : ${JENKINS_FORCE_GET_TARS:="y"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GCE master+node upgrade step 3: run the standard set of non-flaky tests,
+  # using the tests at the version of the code retrieved in step 2.
+  kubernetes-upgrade-gce-step3-e2e)
+    : ${E2E_CLUSTER_NAME:="gce-upgrade"}
+    : ${E2E_NETWORK:="gce-upgrade"}
+    : ${E2E_OPT:="--check_version_skew=false"}
+    : ${E2E_UP:="false"}
+    : ${GINKGO_TEST_ARGS:="--ginkgo.skip=${GCE_DEFAULT_SKIP_TEST_REGEX}|${GCE_FLAKY_TEST_REGEX}"}
+    : ${KUBE_GCE_INSTANCE_PREFIX:="gce-upgrade"}
+    : ${PROJECT:="k8s-jkns-gce-living-upgrades"}
+    ;;
+
+  # GKE
+  # ---
+
+  # Launch a GKE cluster at latest CI, run all non-flaky tests sequentially.
+  kubernetes-e2e-gke-ci)
+    : ${CMD_GROUP:="beta"}
```

Comment:
it seems like several of these vars (CMD_GROUP, DOGFOOD_GCLOUD, maybe others) are shared by all of the GKE jobs - perhaps factor them out?